### PR TITLE
Update V6e TPU Ray Samples

### PR DESCRIPTION
**Pull request by ryanaoleary**
## Why are these changes needed?

This PR updates the v6e TPU Ray CR samples based on feedback from the v6e example guide. This PR also adds a v6e-16 example manifest.

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ryanaoleary** [comment]
cc: @andrewsykim

File: ray-operator/config/samples/ray-job.tpu-v6e-16-multihost.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        template:
+          spec:
+            securityContext:
+              runAsUser: 0
+            containers:
+              - name: ray-worker
+                image: rayproject/ray:2.37.0-py310
+                resources:
+                  limits:
+                    cpu: "100"
```
Comment (andrewsykim) [inline ray-operator/config/samples/ray-job.tpu-v6e-16-multihost.yaml:55]: 24

File: ray-operator/config/samples/ray-job.tpu-v6e-16-multihost.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                    ephemeral-storage: 20Gi
+                    google.com/tpu: "4"
+                    memory: 200G
+                  requests:
+                    cpu: "100"
```
Comment (andrewsykim) [inline ray-operator/config/samples/ray-job.tpu-v6e-16-multihost.yaml:60]: 24

File: ray-operator/config/samples/ray-job.tpu-v6e-16-multihost.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          spec:
+            securityContext:
+              runAsUser: 0
+            containers:
+              - name: ray-worker
+                image: rayproject/ray:2.37.0-py310
+                resources:
+                  limits:
+                    cpu: "100"
+                    ephemeral-storage: 20Gi
```
Comment (andrewsykim) [inline ray-operator/config/samples/ray-job.tpu-v6e-16-multihost.yaml:56]: remove ephemeral storage

File: ray-operator/config/samples/ray-job.tpu-v6e-16-multihost.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                    google.com/tpu: "4"
+                    memory: 200G
+                  requests:
+                    cpu: "100"
+                    ephemeral-storage: 20Gi
```
Comment (andrewsykim) [inline ray-operator/config/samples/ray-job.tpu-v6e-16-multihost.yaml:61]: remove ephemeral stroage

File: ray-operator/config/samples/ray-job.tpu-v6e-256-multihost.yaml [hunk 5]
```diff
@@ -59,6 +61,20 @@ spec:
                     ephemeral-storage: 20Gi
```
Comment (andrewsykim) [inline ray-operator/config/samples/ray-job.tpu-v6e-256-multihost.yaml:61]: update cpu and ephemeral storage here

File: ray-operator/config/samples/ray-cluster.tpu-v6e-16-multihost.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+              requests:
+                cpu: "24"
+                google.com/tpu: "4"
+                memory: 200G
+            env:
+            - name: NODE_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.hostIP
+            - name: VBAR_CONTROL_SERVICE_URL
```
Comment (andrewsykim) [inline ray-operator/config/samples/ray-cluster.tpu-v6e-16-multihost.yaml:61]: Revisit if these env vars are required for v6e in a follow-up PR

**Review by andrewsykim — APPROVED** [review]

File: ray-operator/config/samples/ray-job.tpu-v6e-256-multihost.yaml (hunk 5 above)
Comment (andrewsykim) [inline ray-operator/config/samples/ray-job.tpu-v6e-256-multihost.yaml:61]: One more nit: this example should not include ephemeral-storage limit and use cpu: 24

**Review by kevin85421 — COMMENTED** [review]
1. Many of the YAMLs look similar. Do we need to include all of them in the KubeRay repository?"

2. Do we have documentation explaining how to use these YAMLs? If not, I don't think any users will understand their purpose or use them."

**Comment by andrewsykim** [comment]
@kevin85421 v6 TPU is in public preview and will be GA soon. We plan to update the existing public docs to use v6 but we want to have reference YAML available earlier so it's easier to test it (both manually and with automation)

**Comment by kevin85421** [comment]
@andrewsykim sounds good! Do we have any issues to track the progress?

**Comment by ryanaoleary** [comment]
> TPU is in public previ

I just created an issue for it here: https://github.com/ray-project/ray/issues/48055

**Pull request merged**